File: UnitConvert.java
public class UnitConvert {
    private double curentValue;
    private double result;

    public enum CURENCY {
        USD, AR, RS
    }

    private final double[][] conversionRates = {
        { 1.0       , 4500.0, 83.0      }, // USD to USD, AR, RS
        { 0.00022   , 1.0   , 0.0185    }, // AR to USD, AR, RS
        { 0.012     , 54.0  , 1.0       }  // RS to USD, AR, RS
    };

    public UnitConvert() {
        this.curentValue = 0;
        this.result = 0;
    }

    public double getResult() {
        return result;
    }

    public void convert(double value, CURENCY from, CURENCY to) {
        this.curentValue = value;
        int fromIndex = from.ordinal();
        int toIndex = to.ordinal();
        this.result = curentValue * conversionRates[fromIndex][toIndex];
    }
}

File: unitconvert.c
#include <stdio.h>
typedef enum {
    USD = 0,
    AR = 1,
    RS = 2 
}CURENCY;
double curencyConvert(double amount, int from, int to)
{
    double result = 2;

    double conversion[3][3] = {
        {1.00000, 4500.0, 83.000},   // USD to USD, AR, RS
        {0.00022, 1.0000, 0.0185},  // AR to USD, AR, RS
        {0.01200, 54.000, 1.0000}   // RS to USD, AR, RS
    };

    result = amount * conversion[from][to];
    
    return result;
}

int main()
{
    double amount = 100.0;
    printf("%g USD to AR: %g\n", amount, curencyConvert(amount, USD, AR));
    return 0;
}